server/src/static/js/parent-helper.js

```diff
@@ -2,17 +2,29 @@
 /*global console */
 
 let loaded = false,
-  height = null;
+  height = null,
+  resolveChildReference;
+
+window.childReferencePromise = new Promise((resolve, reject) => {
+  resolveChildReference = resolve;
+});
 
 function doResize() {
   document.getElementById("frame").height = height;
 }
 
 window.onmessage = function(m) {
-  if (m.origin !== location.origin) {
-    console.warn("Parent iframe received message from unexpected origin:", m.origin, "instead of", location.origin);
+  if (m.origin !== POST_MESSAGE_ORIGIN) {
+    console.warn("Parent iframe received message from unexpected origin:", m.origin, "instead of", POST_MESSAGE_ORIGIN);
     return;
   }
+  if (resolveChildReference) {
```

Comment:
Hmm... the way I've done this is with a queue/buffer, which I think is a bit simpler.  So it's something like:

```javascript
function sendToChild(message) {
  if (! sendToChild.childReference) {
    sendToChild.queue.push(message);
    return;
  }
  ... normal postMessage...
}
sendToChild.queue = [];
```

and then we just clear the queue when we get the childReference.  Then code that wants to send a message doesn't have to have any special logic.  (We'll be sending more messages soon, if we show all the clips, show more content inline, allow editing of comments inline, etc)